lib/room.cpp

```diff
@@ -1697,8 +1700,7 @@ void Room::setCanonicalAlias(const QString& newAlias)
 
```

Comment:
When you mentioned in `RoomCanonicalAliasEvent` about compatibility `Room::setCanonicalAlias` you probably did not mean that it would remove all aliases except the canonical one, right? It does exactly that.